File: MarsRover/Java/MarsRover/src/model/scenario/Position.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package model.scenario;

/**
 *
 * @author imanol
 */
public class Position {
    
    public enum Orientation {NORTH,SOUTH, EAST, WEST};

    
    private int coordinateX;
    private int coordinateY;
    private Orientation mapOrientation;

    
    
    public Position(int coordinateX, int coordinateY,Orientation orientation) {
        this.coordinateX = coordinateX;
        this.coordinateY = coordinateY;
        this.mapOrientation = orientation;
    }

    @Override
    public String toString() {
        return "Position{" + "coordinateX=" + coordinateX + ", coordinateY=" + coordinateY + ", mapOrientation=" + mapOrientation + '}';
    }
    
    public Orientation getMapOrientation() {
        return mapOrientation;
    }

    public void setMapOrientation(Orientation mapOrientation) {
        this.mapOrientation = mapOrientation;
    }

    public int getCoordinateX() {
        return coordinateX;
    }

    public void setCoordinateX(int coordinateX) {
        this.coordinateX = coordinateX;
    }

    public int getCoordinateY() {
        return coordinateY;
    }

    public void setCoordinateY(int coordinateY) {
        this.coordinateY = coordinateY;
    }
    
    
    
}

File: MarsRover/JavaScript(TDD)/src/Position.js
var Orientation;
    Orientation = {NORTH:'NORTH', EAST:'EAST',SOUTH:'SOUTH', WEST:'WEST'};

function Position(coordinateX, coordinateY, orientation){

    this.coordinateX = coordinateX;
    this.coordinateY = coordinateY;
    this.orientation = orientation;
}

Position.prototype.getMapOrientation = function() {
    return this.orientation;
}

Position.prototype.setMapOrientation = function(mapOrientation) {
    this.orientation = mapOrientation;
}

Position.prototype.getCoordinateX = function(){
    return this.coordinateX;
}

Position.prototype.setCoordinateX = function(coordinateX) {
    this.coordinateX = coordinateX;
}

Position.prototype.getCoordinateY = function() {
    return this.coordinateY;
}

Position.prototype.setCoordinateY = function(coordinateY) {
    this.coordinateY = coordinateY;
}
Position.prototype.toString = function(coordinateY) {
    return "["+this.coordinateX+","+this.coordinateY+"]";
}